### ISSUE TITLE
Moved 7 albums no problem. 8th and 9th throw errors

### ISSUE THREAD

**Issue by Murdock-droid** [jradwan/Piwigo-physical_photo_move#23]
Hi, 
I'll start with...Thanks for making this plugin! I put off rebuilding my albums because I knew I was going to have to (1) physically move folders around, and then (2) manually update the DB with the new folder numbers. Tonight I updated to the new version of Piwigo, and I decided to check out the list of plugins, and I saw Physical Photo Move...and it was just what I needed. I had my backups from before the piwigo update, so crossed my fingers and tried one album move. It went perfectly. No data was lost. In fact, nothing was lost. So I tried another, and tested again. All good. So I tried another...and another...and another...each time...everything looks good. So I kept rolling. And BAM. Error. The gallery is fine. All moved, and other un-moved albums are fine. The error I am hitting seems to be just with the plugin. And to your credit...no damage to the site, even though I am hitting an error. (Nice.)

I tried to check things here and there, but I wasn't able to find anything. I see what the code is doing on line 454, and it is pretty straightforward, and should work...and it did work...7 times before. I checked permissions, and even set everything in question to 777 just for fun. I am thinking I might have an "ownership" problem, but my host doesn't allow me to view owner. They do allow me to "take ownership", so I did that, and retired. Same error. 

Let me know if you have any ideas. 
Thanks for taking a look, 
John

Warning: rename(./galleries/LanParties/FITES2012,./galleries/LanParties/FITES/FITES2012):
**Permission denied** in /usr/local/4admin/apache/vhosts/njlanparty.com/httpdocs/apps/piwigo/plugins/physical_photo_move/include/functions.inc.php on line 454

Fatal error: Uncaught UnexpectedValueException: DirectoryIterator::__construct(./galleries/LanParties/FITES/FITES2012): failed to open dir: No such file or directory in /usr/local/4admin/apache/vhosts/njlanparty.com/httpdocs/apps/piwigo/plugins/physical_photo_move/include/functions.inc.php:585 Stack trace: #0 /usr/local/4admin/apache/vhosts/njlanparty.com/httpdocs/apps/piwigo/plugins/physical_photo_move/include/functions.inc.php(585): DirectoryIterator->__construct('./galleries/Lan...') #1 /usr/local/4admin/apache/vhosts/njlanparty.com/httpdocs/apps/piwigo/plugins/physical_photo_move/include/functions.inc.php(455): ppm_chmod_r('./galleries/Lan...') #2 /usr/local/4admin/apache/vhosts/njlanparty.com/httpdocs/apps/piwigo/plugins/physical_photo_move/include/functions.inc.php(82): ppm_move_album('73', '69', false) #3 /usr/local/4admin/apache/vhosts/njlanparty.com/httpdocs/apps/piwigo/plugins/physical_photo_move/admin.php(80): ppm_move_item('73', '69', false, 'album') #4 /usr/local/4admin/apache/vhosts/njlanparty.com/httpdocs/a in /usr/local/4admin/apache/vhosts/njlanparty.com/httpdocs/apps/piwigo/plugins/physical_photo_move/include/functions.inc.php on line 585

**Comment by jradwan** [comment]
It does look like a permissions problem of some sort. Does the folder "./galleries/LanParties/FITES/FITES2012" exist yet? Are the other albums you've tried to move with the same error also under "./galleries/LanParties/FITES"? Can you move other albums that are in a different location?

I'm glad you like the plugin and it didn't do any damage to your gallery! ;-)

**Comment by Murdock-droid** [comment]
Thanks for the quick reply.  Yeah. It has to be a permissions problem. I wish I could see that detail. I miss having ROOT to my own server...but external hosting is more reliable than my basement. Anyway, I just tried the move again, hoping maybe there was some silly LOCK in play last night. Not the case. I got the same error. So...I just wanted to finish the move of these last two folders, so I decided to do the file move, and then the DB update by hand. I looked at the variables I had to set, and I updated the record, and moved the folder. And it worked...for the first folder. I went to do the next folder, and I saw the old folder still in the old place. I thought, "Hm...I guess I did COPY in stead of MOVE. Oh well...I will MOVE this one and go back and delete the old folder." And this time...I KNOW I did MOVE. When I got back to the folder...BOTH FOLDERS WERE STILL THERE! It would appear that the MOVE utility the host provides didn't have the permissions to remove the folder after the COPY action. I was able to physically DELETE the folders though...So now they are gone from the original folder, and they exist in the correct folder/album. 

Ok. This can be considered resolved. I have more folders to move. I will let you know if it happens again. I will follow up with my hosting provider first. 

Thanks again for your quick reply. 
John
